src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs

```diff
@@ -646,8 +646,8 @@ private void WriteStreamInfo(string propertyName, IODataStreamReferenceInfo stre
 
             if (this.currentPropertyInfo.IsTopLevel)
             {
-                if (this.JsonLightOutputContext.MessageWriterSettings.LibraryCompatibility <
-                    ODataLibraryCompatibility.Version7 && this.JsonLightOutputContext.MessageWriterSettings.Version < ODataVersion.V401)
+                if ((this.JsonLightOutputContext.MessageWriterSettings.LibraryCompatibility & ODataLibraryCompatibility.WriteTopLevelODataNullAnnotation) == ODataLibraryCompatibility.WriteTopLevelODataNullAnnotation

```

Comment:
Could this if be made a function ? It seems abit complex and is repeated at least once, wrapping it in a function would make it more readable.

Reply:
This are flags checks. I don't think we need a method for that

Reply:
> This are flags checks. I don't think we need a method for that

You don't need to implement a new method. One already exists https://learn.microsoft.com/en-us/dotnet/api/system.enum.hasflag?view=net-8.0. Using it would help shorten the statement and make it easier to understand.

Reply:
Must I use the `HasFlags()` method really?

Reply:
i made this update

Reply:
I thought I had seen a method like that in our code but wasn't sure whether it was an extension or utility method. Since it's already in the base class library, I think we should use it @ElizabethOkerio. It definitely makes the code much more readable and less prone to error.